extensions/amp-analytics/0.1/transport.js

```diff
@@ -118,13 +151,235 @@ export class Transport {
     xhr.send('');
     return true;
   }
+
+  /**
+   * If iframe is specified in config/transport, check whether third-party
+   * iframe already exists, and if not, create it.
+   * @param {!Window} win The window element
+   * @param {!Object<string,string>} transportOptions The 'transport' portion
+   * of the amp-analytics config object
+   * @param {function(!string,
+   *   !../../../src/3p-analytics-common.AmpAnalytics3pResponse)=}
+   *   opt_processResponse An optional function to receive any response
+   *   messages back from the cross-domain iframe
+   */
+  processCrossDomainIframe(win, transportOptions, opt_processResponse) {
+    const frameUrl = dev().assertString(transportOptions['iframe'],
+        'Cross-domain frame parameters missing ${this.type_}');
+    const frameData = this.useExistingOrCreateCrossDomainIframe(
+        win, frameUrl, transportOptions['extraData']);
+    const iframeMessagingClient = frameData.iframeMessagingClient;
+    iframeMessagingClient.registerCallback(
```

Comment:
It's by design that `SubscriptionApi` handles the host side communication, which is to provide the data. IframeMessagingClient is providing a client stub for calling the API. That's to say, the 2 peers are not parity.

There many aspects here, most importantly, `IframeMessagingClient` lives in `3p/` code, it's used by those libs, but never AMP runtime (inabox is an exception, but it's a different runtime).

I'm also not convinced to pursue that as a deadline. I don't see we can achieve anything by just getting this PR into next Tuesday's canary.